saleor/order/tasks.py

```diff
@@ -1,10 +1,27 @@
+import logging
 from typing import List
 
+from celery.exceptions import SoftTimeLimitExceeded
+from django.db.models import Exists, F, Func, OuterRef, Subquery
+from django.db.models.functions import Extract
+from django.utils import timezone
+
 from ..celeryconf import app
+from ..core.tasks import celery_task_lock
+from ..core.tracing import traced_atomic_transaction
+from ..core.utils.events import call_event
+from ..discount.models import Voucher, VoucherCustomer
+from ..payment.models import Payment, TransactionItem
 from ..plugins.manager import get_plugins_manager
+from ..warehouse.management import deallocate_stock_for_orders
+from . import OrderStatus
 from .models import Order
 from .utils import invalidate_order_prices
 
+logger = logging.getLogger(__name__)
+
+ORDER_BATCH_SIZE = 1000
```

Comment:
Please provide doc string with explanation why this is `1000`, have you checked memory usage?